rotkehlchen/tests/unit/test_price_historian.py

```diff
@@ -0,0 +1,159 @@
+from rotkehlchen.history.typing import (
+    DEFAULT_HISTORICAL_PRICE_ORACLES_ORDER,
+    HistoricalPriceOracle,
+)
+from rotkehlchen.typing import Price, Timestamp
+
+
+@pytest.fixture(name='fake_price_historian')
+def fixture_fake_price_historian(historical_price_oracles_order):
+    # NB: custom fixture for quick unit testing. Do not export.
+    # Since this is a singleton and we want it initialized everytime the fixture
+    # is called make sure its instance is always starting from scratch
+    PriceHistorian._PriceHistorian__instance = None
+    price_historian = PriceHistorian(
+        data_directory=MagicMock(spec=Path),
+        cryptocompare=MagicMock(spec=Cryptocompare),
+        coingecko=MagicMock(spec=Coingecko),
+    )
+    price_historian.set_oracles(historical_price_oracles_order)
+    return price_historian
+
+
+def test_all_common_methods_implemented():
+    """Test all historical price oracles implement the expected methods.
+    """
+    for oracle in DEFAULT_HISTORICAL_PRICE_ORACLES_ORDER:
+        if oracle == HistoricalPriceOracle.COINGECKO:
+            instance = Coingecko
+        elif oracle == HistoricalPriceOracle.CRYPTOCOMPARE:
+            instance = Cryptocompare
+        else:
+            raise AssertionError(
+                f'Unexpected historical price oracle: {oracle}. Update this test',
+            )
+
+        # Check 'can_query_history' method exists
+        assert hasattr(instance, 'can_query_history')
+        assert callable(instance.can_query_history)
+        # Check 'query_historical_price' method exists
+        assert hasattr(instance, 'query_historical_price')
+        assert callable(instance.query_historical_price)
+
+
+def test_set_oracles_default_order(fake_price_historian):
+    """Tested via fixture call to 'set_oracles'"""
+    price_historian = fake_price_historian
+
+    expected_oracles = [HistoricalPriceOracle.CRYPTOCOMPARE, HistoricalPriceOracle.COINGECKO]
+    expected_oracle_instances = [price_historian._cryptocompare, price_historian._coingecko]
+
+    assert price_historian._oracles == expected_oracles
+    assert price_historian._oracle_instances == expected_oracle_instances
+
+
+@pytest.mark.parametrize('historical_price_oracles_order', [
+    [HistoricalPriceOracle.COINGECKO],
+])
+def test_set_oracles_custom_order(fake_price_historian):
+    """Tested via fixture call to 'set_oracles'"""
+    price_historian = fake_price_historian
+
+    expected_oracles = [HistoricalPriceOracle.COINGECKO]
+    expected_oracle_instances = [price_historian._coingecko]
+
+    assert price_historian._oracles == expected_oracles
+    assert price_historian._oracle_instances == expected_oracle_instances
+
+
+def test_fiat_to_fiat(fake_price_historian):
+    """Test the price is returned via exchangerates API when requesting the
+    historical price from fiat to fiat.
+    """
+    price_historian = fake_price_historian
+
+    expected_price = Price(FVal('1.25'))
```

Comment:
question: The price here can be anything right? No actual calls happen to any oracle unless I misunderstand something.

Reply:
yes, it can be anything as there are no calls to any other function (otherwise the test would fail as the oracle instances are magic mocks without any return value)

Reply:
all right thanks.